relays/lib-substrate-relay/src/finality/target.rs

```diff
@@ -100,11 +100,11 @@ impl<P: SubstrateFinalitySyncPipeline, TargetClnt: Client<P::TargetChain>>
 		// we can't relay finality if bridge pallet at target chain is halted
 		self.ensure_pallet_active().await?;
 
-		Ok(crate::messages::source::read_client_state::<P::TargetChain, P::SourceChain>(
+		Ok(best_synced_header_id::<P::SourceChain, P::TargetChain>(
```

Comment:
It changes the logic a bit - previous call was returning the best available source header id at **best** target header and now it returns the best available source header id at **best finalized** target header. It is important, because when we are deciding - which source header to submit, we are looking at the **best** state of the target chain, not the **best finalized** state (otherwise our transaction will be obsolete).

Reply:
I missed this. Fixed.